addon/webextension/background/main.js

```diff
@@ -30,10 +30,14 @@ window.main = (function () {
 
   chrome.browserAction.onClicked.addListener(function(tab) {
     if(tab.url.match(/about:(newtab|blank)/i)) {
-      sendEvent("goto-myshots", "about-newtab");
+      catcher.watchPromise(analytics.refreshTelemetryPref().then(() => {
```

Comment:
I specifically considered a preference observer, and it felt much more complicated to me without much benefit.  The bootstrap code has to initiate the message and it causes persistent code in the addon to watch things (where in this version nothing is observed when the add-on is passive).  This seems reasonably respectful, since it only gets it wrong for a small amount of time if you happen to change the pref in the middle of a Page Shot session, and the impact is very limited.